internal/juju/applications.go

```diff
@@ -248,10 +248,12 @@ func (c applicationsClient) CreateApplication(input *CreateApplicationInput) (*C
 		return nil, fmt.Errorf("expected only one resolution, received %d", len(resolved))
 	}
 	resolvedCharm := resolved[0]
-
 	if resolvedCharm.Error != nil {
 		return nil, resolvedCharm.Error
 	}
+	if resolvedCharm.Origin.Type == "bundle" {
+		return nil, jujuerrors.Errorf("%q is a bundle not a charm", input.CharmName)
```

Comment:
the message might be a bit nicer imo, but it still gives the info, so feel free to merge as is

```suggestion
		return nil, jujuerrors.Errorf("unable to deploy bundles in the terraform provider, expected a charm, given bundle : %q", input.CharmName)
```

Reply:
I was debating, the full printed message to the user is
```
Unable to create application, got error: "charmed-kubernetes" is a bundle not a charm
```